spec/s3.spec.js

```diff
@@ -0,0 +1,192 @@
+    var AwsSdkMock, promiseMock, s3;
+    
+    /**
+     * Checks the result for the promise object passed in.
+     *
+     * @param {Object} results
+     * @param {boolean} successful
+     * @param {string} message
+     */
+    function checkPromiseResults(results, successful, message) {
+        expect(results.success).toBe(successful);
+        expect(results.value).toEqual(message);
+    }
+    
+    beforeEach(() => {
+        var s3File;
+        
+        s3File = require("../lib/storage/s3");
+        promiseMock = require("./mock/promise-mock");
+        class S3Fake {
+            constructor(params) {
+                this.params = params;
+                this.deleteObjectAsync = jasmine.createSpy("deleteObjectAsync");
+                this.getObjectAsync = jasmine.createSpy("getObjectAsync");
+                this.listObjectsAsync = jasmine.createSpy("listObjectsAsync");
+                this.putObjectAsync = jasmine.createSpy("putObjectAsync");
+            }
+        }
+        AwsSdkMock = {
+            "S3": S3Fake,
+            "config": {
+                "region": null
+            }
+        };
+        s3 = new s3File(AwsSdkMock, promiseMock);
+    });
+    describe("configure", () => {
+        beforeEach(() => {
+            AwsSdkMock.S3 = jasmine.createSpy("s3Mock");
+        });
+        it("passes in configuration options for all", () => {
+            s3.transit();
+            expect(s3.aws.config.region).toBe("us-east-1");
+            expect(AwsSdkMock.S3).toHaveBeenCalledWith({params: {
+                Bucket: null
+            }});
+        });
+        it("passes in configuration options for all", () => {
+            s3.configure({
+                region: "us-west-1",
+                bucket: "test-bucket"
+            });
+            s3.transit();
+            expect(s3.aws.config.region).toBe("us-west-1");
+            expect(AwsSdkMock.S3).toHaveBeenCalledWith({params: {
+                Bucket: "test-bucket"
+            }});
+        });
+    });
+    describe("fileDel", () => {
+        var transit;
+        
+        beforeEach(() => {
+            transit = s3.transit();
+            transit.deleteObjectAsync.andCallFake((params) => {
+                if (params.Key == "afile") {
+                    return promiseMock.resolve("deleted file");
+                }
+
+                return promiseMock.reject("error deleting file");
+            });
+        });
+        it("deletes a file", () => {
+            checkPromiseResults(s3.fileDel("afile"), true, "deleted file");
+        });
+    });
+    describe("fileGet", () => {
+        var transit;
+        
+        beforeEach(() => {
+            transit = s3.transit();
+            transit.getObjectAsync.andCallFake((params) => {
+                if (params.Key == "afile") {
+                    return promiseMock.resolve("got data");
+                }
+
+                return promiseMock.reject("error fetching file");
+            });
```

Comment:
This is starting to get redundant.  You might want to just set up mocks in the initial `beforeEach()` for all of the tests.

```
class S3Fake {
    constructor(params) {
        this.params = params;
        [
            'deleteObjectAsync',
            'getObjectAsync',
            'listObjectAsync',
            'putObjectAsync'
        ].forEach((method) => {
            this[method] = jasmine.createSpy(method);
            this[method].andCallFake((params) => {
                return promiseMock.resolve(params);
            });
        }
    }
}
```

You don't test any failed promises, which is ok by me, so the fake methods attached just resolve the promise with the arguments.  With that in place you can check the arguments.

```
expect(s3.fileList().status()).toEqual({
    success: true,
    value: {
        Prefix: "accounts"
    }
});
```